pkg/snapshot/README.md

```diff
@@ -0,0 +1,203 @@
+- `<TYK_AUTH>`: Operator user API Key.
+- `<TYK_ORG>`: Operator user ORG ID.
+
+> For more details on how to obtain the URL and credentials, please visit [Tyk Docs](https://tyk.io/docs/tyk-stack/tyk-operator/installing-tyk-operator/#tyk-self-managed-hybrid).
+
+
+
+### Exporting API Definitions
+
+#### Specify Category to export
+
+By default, tyk-operator exports ApiDefinitions defined in `#operator` category. You can change default category via `--category` flag.
+
+```bash
+TYK_MODE=pro TYK_URL=<TYK_URL> TYK_AUTH=<TYK_AUTH> TYK_ORG=<TYK_ORG> ./tyk-operator --snapshot output.yaml --category k8s
+```
+
+You can also dump all ApiDefinitions from Dashboard via `--all` flag.
+
+```bash
+TYK_MODE=pro TYK_URL=<TYK_URL> TYK_AUTH=<TYK_AUTH> TYK_ORG=<TYK_ORG> ./tyk-operator --snapshot output.yaml --all
+```
+
+#### Output CR
+
+`tyk-operator` CLI creates an output file specified via `--snapshot` flag. Each ApiDefinition CR metadata has a default name  `replace-me-{i}` where `{i}` increases by each ApiDefinition.
+
+For example,
+```yaml
+apiVersion: tyk.tyk.io/v1alpha1
+kind: ApiDefinition
+metadata:
+  name: replace-me-0 # Default name for ApiDefinition CRs
+  namespace: default # Default namespace for ApiDefinition CRs
+spec:
+  ...
+```
+
+In order to specify CR metadata, you can use Config Data. For specified ApiDefinitions,
+snapshot CLI generates ApiDefinition CRs based on Config Data of that specific ApiDefinition.
+
+```json
+{
+  "k8sName": "metadata-name",
+  "k8sNs": "metadata-namespace"
+}
+```
+
+The CLI checks for `k8sName` and `k8sNs` fields of each ApiDefinition's Config Data 
+to generate metadata of the output CR. If these fields exist, the CLI uses the 
+values specified in these fields. Otherwise, it uses default values (`replace-me-` 
+for name and `default` for namespace) for them.
+
+Assume we have the following ApiDefinitions, two of which are categorized as `#testing` 
+and created on our Dashboard.
+
+![apis](./img/apis.png)
+
+If we would like to specify metadata of the `test-api-5`, we can update Config Data
+of the ApiDefinition, as follows.
+
+![config-data](./img/config-data.png)
+
+So, the generated output for this environment will look as follows;
+```bash
+TYK_MODE=pro TYK_URL=<TYK_URL> TYK_AUTH=<TYK_AUTH> TYK_ORG=<TYK_ORG> ./tyk-operator --snapshot output.yaml --category testing
+```
+```yaml
+# output.yaml
+
+apiVersion: tyk.tyk.io/v1alpha1
+kind: ApiDefinition
+metadata:
+  creationTimestamp: null
+  name: httpbin-api-5   # obtained from Config Data's "k8sName" field.
+  namespace: staging    # obtained from Config Data's "k8sNs" field. 
+spec:
+  name: 'test-api-5 #testing'
+  ...
+---
+apiVersion: tyk.tyk.io/v1alpha1
+kind: ApiDefinition
+metadata:
+  creationTimestamp: null
+  name: replace-me-1    # Since Config Data does not include "k8sName", default name is used.
+  namespace: default    # Since Config Data does not include "k8sKey", default namespace is used.
```

Comment:
what is `k8sKey`?

Reply:
sorry, it should be `k8sNamespace`. I will update it. Thanks for spotting it